pkg/services/notifications/smtp.go

```diff
@@ -0,0 +1,147 @@
+package notifications
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util/errutil"
+	gomail "gopkg.in/mail.v2"
+)
+
+type SmtpClient struct {
+	cfg setting.SmtpSettings
+}
+
+func ProvideSmtpService(cfg *setting.Cfg) (Mailer, error) {
+	return NewSmtpClient(cfg.Smtp)
+}
+
+func NewSmtpClient(cfg setting.SmtpSettings) (*SmtpClient, error) {
+	client := &SmtpClient{
+		cfg: cfg,
+	}
+
+	return client, nil
+}
+
+func (sc *SmtpClient) Send(messages ...*Message) (int, error) {
```

Comment:
The code below this point is not new - it was moved here from `mailer.go`. Idea being to make a distinction between the notification service's app logic and the actual mechanics of sending SMTP requests.